src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/FlowLayoutPanelDesigner .cs

```diff
@@ -637,10 +636,9 @@ private void ReorderControls(DragEventArgs de)
             else
             {
                 // We are doing a copy, so let's copy the controls.
-                List<Control> tempList = new();
-                tempList.AddRange(_dragControls);
+                List<IComponent> tempList = [.. _dragControls];
 
-                DesignerUtils.CopyDragObjects(tempList, Component.Site);
+                tempList = DesignerUtils.CopyDragObjects(tempList, Component.Site);
```

Comment:
```suggestion
                List<IComponent> tempList = DesignerUtils.CopyDragObjects(_dragControls, Component.Site);
```
and remove the line above?

Reply:
I am unsure what is correct. Other usages have a similar pattern of creating a temp list and passing it in. But I think they could be wrong as well.

Reply:
Just tested it in DemoConsole and it looks like it works fine. So we will go with that.

Reply:
@halgab did you want to investigate the rest of the usages of `DesignerUtils.CopyDragObjects` to make sure they aren't creating a list unnecessarily?

Reply:
I've just checked, and it was the only such occurence. `DesignerUtils.CopyDragObjects` always returns a new list by the way. What I have my eye on is changing this method's signature to accept something more strongly typed than `ICollection`. Moving `ToolStripItemDataObject.DragComponents` off `ArrayList` is a prerequisite though.

Reply:
Scratched my own itch: #10626

Reply:
I had a quick look, it seems possible to narrow everything to `IEnumerable<IComponent>`, but `CopyDragObjects ` would need a little bit of refactoring to avoid calling `Count()` multiple times.

`ToolStripItemDataObject.DragComponents` might be an issue depending on serialization, it will need to be tested. I suggest that change goes into its own PR gets thoroughly tested.